### PR TITLE
`unsplash-js`: improve types with stricter types

### PR DESCRIPTION
**Pull request by lukechesser**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Changelog](https://github.com/unsplash/unsplash-js/blob/master/CHANGELOG.md#620) and [API docs](https://unsplash.com/documentation)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/unsplash-js/index.d.ts [hunk 1]
```diff
@@ -7,6 +7,19 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.2
 
+export type PhotoOrderBy = 'latest' | 'oldest';
+export type Orientation = 'portrait' | 'landscape' | 'squareish';
+export type ContentSafety = 'high' | 'low';
+export type CollectionIds = ReadonlyArray<number> | ReadonlyArray<string>;
```
Comment (OliverJAsh) [inline types/unsplash-js/index.d.ts:13]: If you want to allow a mixture of string and number in the same array, you could make this `ReadonlyArray<string | number>`.

**Review by OliverJAsh — APPROVED** [review]
🎉

File: types/unsplash-js/index.d.ts (hunk 1 above)
Comment (lukechesser) [reply types/unsplash-js/index.d.ts:13]: Ah that's actually what I wanted to do, thanks!

Comment (lukechesser) [inline types/unsplash-js/index.d.ts:13]:
```suggestion
export type CollectionIds = ReadonlyArray<number | string>;
```

**Review by OliverJAsh — APPROVED** [review]

**Comment by samijaber** [comment]
Ready to merge

**Pull request merged**